barretenberg/cpp/src/barretenberg/proof_system/execution_trace/execution_trace.cpp

```diff
@@ -85,52 +88,52 @@ typename ExecutionTrace_<Flavor>::TraceData ExecutionTrace_<Flavor>::construct_t
     return trace_data;
 }
 
-template <class Flavor>
-std::vector<typename ExecutionTrace_<Flavor>::TraceBlock> ExecutionTrace_<Flavor>::create_execution_trace_blocks(
-    const Builder& builder)
+template <class Flavor> void ExecutionTrace_<Flavor>::populate_public_inputs_block(Builder& builder)
 {
-    std::vector<TraceBlock> trace_blocks;
-
-    // Make a block for the zero row
-    if constexpr (Flavor::has_zero_row) {
-        TraceBlock zero_block;
-        for (auto& wire : zero_block.wires) {
-            wire.emplace_back(builder.zero_idx);
-        }
-        for (auto& selector : zero_block.selectors.get()) {
-            selector.emplace_back(0);
-        }
-        trace_blocks.emplace_back(zero_block);
-    }
-
-    // Make a block for the ecc op wires
-    if constexpr (IsGoblinFlavor<Flavor>) {
-        trace_blocks.emplace_back(builder.ecc_op_block);
-    }
-
-    // Make a block for the public inputs
-    TraceBlock public_block;
+    // Update the public inputs block
     for (auto& idx : builder.public_inputs) {
         for (size_t wire_idx = 0; wire_idx < NUM_WIRES; ++wire_idx) {
             if (wire_idx < 2) { // first two wires get a copy of the public inputs
-                public_block.wires[wire_idx].emplace_back(idx);
+                builder.blocks.pub_inputs.wires[wire_idx].emplace_back(idx);
             } else { // the remaining wires get zeros
-                public_block.wires[wire_idx].emplace_back(builder.zero_idx);
+                builder.blocks.pub_inputs.wires[wire_idx].emplace_back(builder.zero_idx);
             }
         }
-        for (auto& selector : public_block.selectors.get()) {
+        for (auto& selector : builder.blocks.pub_inputs.selectors) {
             selector.emplace_back(0);
         }
     }
+    builder.blocks.pub_inputs.is_public_input = true;
```

Comment:
Could you elaborate (in comments) what this does? It seems intuitive that pub_inputs is public input

Reply:
This is no longer needed since the pub inputs update Pr that you recently reviewed. I'll remove